libtenzir/builtins/operators/every_cron.cpp

```diff
@@ -194,20 +206,23 @@ class every_operator final : public operator_base {
   }
 
   auto name() const -> std::string override {
-    return "every";
+    return std::string{scheduler_.name};
   }
 
-  friend auto inspect(auto& f, every_operator& x) -> bool {
-    return f.object(x).fields(f.field("op", x.op_),
-                              f.field("interval", x.interval_));
+  friend auto inspect(auto& f, scheduled_execution_operator& x) -> bool {
+    return f.object(x).fields(f.field("op", x.op_), x.scheduler_.field(f));
   }
 
 private:
   operator_ptr op_;
-  duration interval_;
+  Scheduler scheduler_;
 };
 
-class every_plugin final : public virtual operator_plugin<every_operator> {
+/// Base plugin template for scheduled execution operators
+/// The actual parsing is handled by the `Scheduler` type
+template <scheduler_concept Scheduler>
+class scheduled_execution_plugin
+  : public virtual operator_plugin<scheduled_execution_operator<Scheduler>> {
```

Comment:
Again, it would be great if you could use proper punctuation, unless the line acts as a headline.